src/control-plane/backend/lambda/api/router/segment.ts

```diff
@@ -212,17 +211,17 @@ function commonValidationsForSegment() {
                   body(path + 'hasDone').isBoolean(),
                   body(path + 'event.eventName').notEmpty(),
                   body(path + 'event.operator').optional().isIn(['and', 'or']),
-                  body(path + 'event.eventParameterConditions.*.parameterType').isIn(Object.values(ParameterType)),
+                  body(path + 'event.eventParameterConditions.*.parameterType').isIn(Object.values(MetadataSource)),
                   body(path + 'event.eventParameterConditions.*.parameterName').notEmpty(),
-                  body(path + 'event.eventParameterConditions.*.dataType').isIn(Object.values(ParameterDataType)),
-                  body(path + 'event.eventParameterConditions.*.conditionOperator').isIn(Object.values(ConditionOperator)),
+                  body(path + 'event.eventParameterConditions.*.dataType').isIn(Object.values(MetadataValueType)),
+                  body(path + 'event.eventParameterConditions.*.conditionOperator').isIn(Object.values(ExploreAnalyticsOperators)),
                   body(path + 'event.eventParameterConditions.*.inputValue').custom(validateParameterConditionInputValue),
                   body(path + 'metricCondition.metricType').isIn(Object.values(SegmentFilterEventMetricType)),
-                  body(path + 'metricCondition.conditionOperator').isIn(Object.values(ConditionNumericOperator)),
```

Comment:
`ConditionNumericOperator` is different from `ConditionOperator` (`ExploreAnalyticsOperators`), it's only applied to numeric variables.